app/scripts/lib/mailcheck.js

```diff
@@ -69,14 +69,20 @@ function (Tooltip, Url) {
 
           tooltip.render();
 
-          tooltip.$el.on('click', 'span', function () {
-            element.val(suggestion.full);
-            // the user has used the suggestion
-            metrics.logEvent('tooltip.mailcheck-used');
-            tooltip._destroy();
+          tooltip.$el.on('click keypress', 'span', function (e) {
+            // if a click event is triggered or an enter key is pressed, destroy
+            // the tooltip.
+            if (e.type === 'click' || e.which === 13) {
+              element.val(suggestion.full);
+              // the user has used the suggestion
+              metrics.logEvent('tooltip.mailcheck-used');
+              tooltip._destroy();
+            }
           });
         }
       });
     }
+
+    element[0].previousValue =  element.val();
```

Comment:
Could we reinforce this and add a check if `var element = $(target);` found an element in `element[0]`